lib/travis/build/templates/header.sh

```diff
@@ -155,64 +155,31 @@ travis_terminate() {
   exit $1
 }
 
+_write_visual_bells() {
+  local minutes=$1
+  local tick=0
+
+  while [[ $tick -lt $minutes ]]; do
+    >&2 echo -en "\a"
+    sleep 60
+    let tick=$tick+1
+  done
+}
+
 travis_wait() {
   local timeout=$1
 
   if [[ $timeout =~ ^[0-9]+$ ]]; then
-    <%# looks like an integer, so we assume it's a timeout %>
+    <%# looks like an integer, so we assume it is a timeout %>
     shift
   else
     <%# default value %>
     timeout=20
   fi
 
-  local cmd="$@"
-  local log_file=travis_wait_$$.log
-
-  $cmd &>$log_file &
-  local cmd_pid=$!
-
-  travis_jigger $! $timeout $cmd &
-  local jigger_pid=$!
-  local result
-
-  {
-    wait $cmd_pid 2>/dev/null
-    result=$?
-    ps -p$jigger_pid &>/dev/null && kill $jigger_pid
-  }
-
-  if [ $result -eq 0 ]; then
-    echo -e "\n${ANSI_GREEN}The command $cmd exited with $result.${ANSI_RESET}"
-  else
-    echo -e "\n${ANSI_RED}The command $cmd exited with $result.${ANSI_RESET}"
-  fi
-
-  echo -e "\n${ANSI_GREEN}Log:${ANSI_RESET}\n"
-  cat $log_file
-
-  return $result
-}
-
-travis_jigger() {
-  <%# helper method for travis_wait() %>
-  local cmd_pid=$1
-  shift
-  local timeout=$1 <%# in minutes %>
-  shift
-  local count=0
-
-  <%# clear the line %>
-  echo -e "\n"
-
-  while [ $count -lt $timeout ]; do
-    count=$(($count + 1))
-    echo -ne "Still running ($count of $timeout): $@\r"
-    sleep 60
-  done
+  _write_visual_bells $timeout &
 
-  echo -e "\n${ANSI_RED}Timeout (${timeout} minutes) reached. Terminating \"$@\"${ANSI_RESET}\n"
-  kill -9 $cmd_pid
+  travis_cmd "$*"
```

Comment:
`travis_cmd` needs to take extra flags at the end, so that it can do other things echo command it is executing, set timing markers, etc.

`$*` is the right invocation here.